src/Database/Query.php

```diff
@@ -492,10 +492,11 @@ public function from($tables = [], $overwrite = false)
      * @param array|string|null $tables list of tables to be joined in the query
      * @param array $types associative array of type names used to bind values to query
      * @param bool $overwrite whether to reset joins with passed list or not
+     * @param bool $subquery whether this is a subquery
      * @see \Cake\Database\Type
      * @return $this
      */
-    public function join($tables = null, $types = [], $overwrite = false)
+    public function join($tables = null, $types = [], $overwrite = false, $subquery = false)
```

Comment:
This signature change is not acceptable to me, and looks like a backwards incompatible change